lib/matplotlib/cbook/__init__.py

```diff
@@ -2033,6 +2034,24 @@ def _setattr_cm(obj, **kwargs):
     """
     sentinel = object()
     origs = [(attr, getattr(obj, attr, sentinel)) for attr in kwargs]
+    # When you access a Python method the function is bound
+    # to the object at access time so you get a new instance
+    # of MethodType every time so replace them with sentinel
+    #
+    # https://docs.python.org/3/howto/descriptor.html#functions-and-methods
+    def filter_restores(obj, attr, val):
+        if inspect.ismethod(val):
```

Comment:
this check is redundant with the check below (it won't be in the object dict) and will in fact be wrong in the obscure case where you patch an object's attribute using some *other* object's bound method, e.g.
```python
class C:
    def meth(self): pass

class D:
    pass

d = D()
d.foo = C().meth
```

Reply:
Great!  I am never happy when we have to resort to using the inspect module.